app/models/concerns/race_validator.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+# Validates that a given date (attribute) is valid
+class RaceValidator < ActiveModel::Validator
+  def validate(record)
+    # If value is unsuccessfully typecast to a date, it will be nil, so validate on the value before cast
+    has_non_exclusive_race = record.white || record.black_or_african_american || record.american_indian_or_alaska_native ||
+                             record.asian || record.native_hawaiian_or_other_pacific_islander || record.race_other
+
+    return unless has_non_exclusive_race ? (record.race_unknown || record.race_refused_to_answer) : (record.race_unknown && record.race_refused_to_answer)
```

Comment:
The first case (where `has_non_exclusive_race` is `true`) uses an `||` because race cannot be a non_exclusive_race + either one of the exclusive ones, the second case uses a `&&` because none of the non_exclusive_races are selected, which means it's only invalid if both `unknown` and `refused_to_answer` are selected, but it's valid if only one of them are selected. Does that make sense or am I missing something?

Reply:
Yep - mistake on my part. Thanks!